dask_ml/model_selection/_hyperband.py

```diff
@@ -0,0 +1,323 @@
+                      if r['bracket_iter'] == result['bracket_iter'] and
+                      r['bracket'] == result['bracket']]
+
+    if not asynchronous:
+        bracket_completed = len(bracket_models) == result['num_models']
+        if bracket_completed:
+            to_keep = len(bracket_models) // eta
+            if to_keep <= 1:
+                return []
+            top_results = _top_k(bracket_models, k=to_keep, sort="score")
+            for job in top_results:
+                job["num_models"] = to_keep
+                job['partial_fit_calls'] *= eta
+                job['bracket_iter'] += 1
+            return top_results
+        return []
+    raise ValueError
+
+
+def _create_model(model, params, random_state=42):
+    model = clone(model).set_params(**params)
+    if 'random_state' in model.get_params():
+        model.set_params(random_state=random_state)
+    #  meta = {}  # right now no meta information
+    return model, None
+
+
+def _bracket_name(s):
+    return "bracket={s}".format(s=s)
+
+
+def _model_id(s, n_i):
+    return "bracket={s}-{n_i}".format(s=s, n_i=n_i)
+
+
+async def _hyperband(model, params, X, y, max_iter=None, eta=None,
+                     dry_run=False, fit_params={}, random_state=42,
+                     asynchronous=None):
+    client = default_client()
+    rng = check_random_state(random_state)
+    N, R, brackets = _get_nr(max_iter, eta=eta)
+    params = iter(ParameterSampler(params, n_iter=sum(N),
+                                   random_state=random_state))
+    if isinstance(model, Incremental):  # this should probably be handled elsewhere
+        model = model.estimator
+    params = {_model_id(s, n_i): next(params)
+              for s, n, r in zip(brackets, N, R) for n_i in range(n)}
+    model_futures = {_model_id(s, n_i):
+                     client.submit(_create_model, model,
+                                   params[_model_id(s, n_i)],
+                                   random_state=random_state)
+                     for s, n, r in zip(brackets, N, R) for n_i in range(n)}
+
+    # lets assume everything in fit_params is small and make it concrete
+    fit_params = await client.compute(fit_params)
+
+    r = train_test_split(X, y, test_size=0.15, random_state=random_state)
+    X_train, X_test, y_train, y_test = r
+    if isinstance(X, da.Array):
+        X_train = futures_of(X_train.persist())
+        # X_test = futures_of(X_test.persist())  # if we should operate in chunks
+        X_test = client.compute(X_test)  # if we should pass a single chunk
+    else:
+        X_train, X_test = await client.scatter([X_train, X_test])
+    if isinstance(y, da.Array):
+        y_train = futures_of(y_train.persist())
+        # y_test = futures_of(y_test.persist())  # if we should operate in chunks
+        y_test = client.compute(y_test)  # if we should pass a single chunk
+    else:
+        y_train, y_test = await client.scatter([y_train, y_test])
```

Comment:
Should this behavior be configurable?

Reply:
Hrm, I may have meant to say `x_test, y_test` here rather than `y_train`, which should presumably be a dask.array-type thing.

Reply:
I think the variables are correct. `X_train` and `X_test` were already scattered above.

I was wondering whether the scattering should be configurable, but I was mistaken on the default `broadcast` behavior. This seems fin as is.

Reply:
I can imagine reasons to pass in a broadcast= keyword.

Reply:
The default `broadcast` behavior is still being used in https://github.com/dask/dask-ml/blob/ce81ad12b156548ca31b447bf34ab1ba119e9469/dask_ml/model_selection/_incremental.py#L148

Should it be kept that way?